pkg/storage/chunk/predicate.go

```diff
@@ -1,16 +1,22 @@
 package chunk
 
 import (
+	"github.com/grafana/loki/pkg/querier/plan"
 	"github.com/prometheus/prometheus/model/labels"
-
-	"github.com/grafana/loki/pkg/logql/syntax"
 )
 
 type Predicate struct {
 	Matchers []*labels.Matcher
-	Filters  []syntax.LineFilter
+	plan     *plan.QueryPlan
```

Comment:
Matchers and plan now contain redundant information. The plan already contains the matchers.

Reply:
As discussed in Slack. We will leave as is right now. We can consider refactoring that on a separate PR.